tail/tail.go

```diff
@@ -107,6 +109,66 @@ type Tailer struct {
 	offset      int // Bytes read within the current reader.
 }
 
+type checkpointRef struct {
+	name  string
+	index int
+}
+
+// listCheckpoints used to ignore all checkpoints w/ .tmp extensions, this
+// is fine for most cases. In the case of the sidecar starting up, it's possible
+// that prometheus has decided to trigger a checkpoint before the sidecar has
+// caught up. This causes all sorts of problems w/ series references missing. To
+// address this, the sidecar must check if a checkpoint is underway
```

Comment:
Will you add example filenames that a user might see, if they inspected `ReadDir()` output? What do a WAL segment, a finished checkpoint, and an in-progress checkpoint look like? I admit the test has these, and I know it, but I'd like to read it here or somewhere nearby the call to `strings.Split()`.

Then, I think it's worth saying why it doesn't matter how many old checkpoints there are, only the last checkpoint is ever interesting, and that in fact we expect to see stale, abandoned checkpoints lying around.

Reply:
Added more details here.